sky/spot/spot_state.py

```diff
@@ -100,19 +103,22 @@ class SpotStatus(enum.Enum):
     # Terminal statuses
     # SUCCEEDED: The job is finished successfully.
     SUCCEEDED = 'SUCCEEDED'
+    # CANCELLED: The job is cancelled by the user.
+    CANCELLED = 'CANCELLED'
     # FAILED: The job is finished with failure from the user's program.
     FAILED = 'FAILED'
     # FAILED_SETUP: The job is finished with failure from the user's setup
     # script.
     FAILED_SETUP = 'FAILED_SETUP'
+    # FAILED_CONFIG: The job is finished with failure because of invalid
+    # job configuration.
+    FAILED_CONFIG = 'FAILED_CONFIG'
```

Comment:
It is not immediately clear that name-too-long is a "config" problem. How about `FAILED_OTHER_REASON`, or something better?